frontend/src/stores/user.js

```diff
@@ -57,36 +73,26 @@ class UserStore {
     };
 
     async authenticate() {
-        // code try to authenticate...
-
-        console.log('Beep beep.... I am authenticating');
+        this.isAuthenticating = true;
+        const pin = toStringFromObject(this.pinCode);
+        const phoneNumber = processPhoneNumber(this.phoneNumber);
         try {
-            await new Promise((resolve, reject) => {
-                // this is because "this" loses scope in setTimeOut
-                const execute = () => {
-                    const processedPhoneNumber = this.phoneNumber
-                        .replace(/\s/g, '')
-                        .replace('0', '+358');
-                    const proccessedPinCode = toStringFromObject(this.pinCode);
-
-                    if (
-                        fakeCredentials.phoneNumber === processedPhoneNumber &&
-                        fakeCredentials.pinCode === proccessedPinCode
-                    ) {
-                        resolve('done');
-                    } else {
-                        reject('Incorrect credentials');
-                    }
-                };
-                setTimeout(execute.bind(this), 1000);
+            const response = await fetch(`/api/users/login`, {
```

Comment:
Could you please implement a wrapper for fetch as it is quite verbose to set the headers for each request and map to json.

Reply:
I will make a ticket.